docs/isl-2-0/spec.md

```diff
@@ -0,0 +1,668 @@
+
+The `distinct` annotation on the type reference argument indicates that the constraint should not allow repeated elements within the container.
+For the purpose of this constraint, the comparison of the values (including any annotations) in the container is governed by the equivalence rules defined by the [Ion data model](https://amzn.github.io/ion-docs/docs/spec.html#the-ion-data-model).
+
+{% comment %}{% include example.md title="Disallowing duplicate elements" code_file="examples/placeholder.isl" %}{% endcomment %}
+
+## exponent
+
+{% include grammar-element.md productions="exponent" %}
+
+This constraint specifies an exact or minimum/maximum range indicating the exponent of the Ion decimal.
+Remember that decimal values with digits after the decimal point have a _negative_ exponent, so to require at least two digits after
+
+{% include note.html type="tip" content="See [Modeling SQL Decimals](../cookbook/sql-decimals.md) to learn how `exponent` relates to the concept of _scale_." %} 
+
+{% comment %}{% include example.md title="`exponent` exact" code_file="examples/placeholder.isl" %}{% endcomment %}
+{% comment %}{% include example.md title="`exponent` range" code_file="examples/placeholder.isl" %}{% endcomment %}
+
+### Converting from Ion Schema 1.0 `scale` to `exponent`
+
+The `exponent` constraint replaces the `scale` constraint found in ISL 1.0.
+To convert a from `scale` to `exponent`, simply multiply a single argument by -1 (e.g. `scale: 4` becomes `exponent: -4`) or reflect the range relative to 0 (e.g. `scale: range::[min, 5]` becomes `exponent: range::[-5, max]`).
+
+## field_names
+
+{% include grammar-element.md productions="field_names" %}
+
+Applicable for `struct` values.
+
+The field_names constraint defines the type and/or constraints for all field names within a struct.
+Field names are symbols and will be represented as symbol values for the purpose of validation.
+Any value that is not a struct as well as `null.struct` will always be invalid for this constraint.
+
+{% comment %}{% include example.md title="Limiting the length of field names" code_file="examples/placeholder.isl" %}{% endcomment %}
+{% comment %}{% include example.md title="Limiting the characters used in field names" code_file="examples/placeholder.isl" %}{% endcomment %}
+
+The `distinct` annotation on the type reference argument indicates that the constraint should not allow field names to be repeated within a single struct.
+
+{% comment %}{% include example.md title="Disallowing repeated field names" code_file="examples/placeholder.isl" %}{% endcomment %}
+
+## fields
+
+{% include grammar-element.md productions="fields,field" %}
+
+Declares one or more field constraints of a struct.
+
+Field names defined for a particular struct type shall be unique.
+A field may narrow its declared type by specifying additional constraints. 
+By default, a field is constrained by `occurs: optional`.
```

Comment:
Just would like to confirm that the constraint 'occurs' in the field could not be overridden right ? Since the field name shall be unique. Is 'occurs: 1' acceptable here?